test_vm/vendor/cookbooks/dbfit_test/metadata.rb

```diff
@@ -7,7 +7,7 @@
 version          '0.3.0'
 depends          'mysql2_chef_gem'
 depends          'yum-mysql-community'
-depends          'mysql', '~> 8.2'
+depends          'mysql'
```

Comment:
`~> 8.2` is pessimistic locking and means >= 8.2 and < 9.0. So `8.3.1` meets the constraint.

> So why would we now want to keep this constraint?

Well, I don't know should we keep it or not. The old setting has the constraint, I guess we did it with idea that:
* It reminds us that versions < 8.2 are problematic and prevents us from using them.
* Also it would not automatically jump to 9.0 which might have breaking changes (mysql cookbook has proven to be a bit brittle).

Essentially the question is do we prefer `~> 8.2` or `>= 8.2`. The latter will give us signal that there is version >= 9.0. Maybe that's not a bad thing - we're anyway locking the versions with the `Berksfile.lock` and testing if things still work OK upon changing it.

Up to your judgement whether to remove the dependency constraint. It might be better to have it as separate PR since I believe it doesn't contribute to fixing the broken provisioning this time.